controllers/gateway/resource.go

```diff
@@ -143,7 +172,7 @@ func (ctx *gatewayContext) createGatewayDeployment() (*appv1.Deployment, error)
 	if err != nil {
 		return nil, err
 	}
-	return ctx.controller.k8sClient.AppsV1().Deployments(deployment.Namespace).Create(deployment)
+	return ctx.controller.k8sClient.AppsV1().Deployments(ctx.gateway.Namespace).Create(deployment)
```

Comment:
Auto generated Deployment and Service should be in the same namespace as Gateway, it doesn't make sense to create objects in a namespace other than its own.

Reply:
👍